### PR TITLE
Ftr 4087 materials redesign

### LINKED ISSUE #4087: Workspace redesign
Epic for workspace redesign issues.

### PR DESCRIPTION
**Pull request by jjlankinen**
Second coming of Muikku redesign. 

Closes #4087

### CODE REVIEW

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/material-page.scss [hunk 1]
```diff
@@ -0,0 +1,2082 @@
...
+
+  @include breakpoint($breakpoint-pad) {
+    margin: 20px 0;
+    padding: 0 20px;
+  }
+}
+
+.material-page__content--view-restricted {
+  align-items: center;
+  color: darken($color-disabled, 30%);
```
[the remark this answers is not in the document]
Comment (jjlankinen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/elements/material-page.scss:95]: This seems weird error as color property IS using variable called $color-disabled

We use darken functions in many places and this seems to be the only place where scss-lint gets triggered. Other places uses background-color though.

File: muikku-core-plugins/src/main/resources/META-INF/resources/jsf/workspace/structure.txt [hunk 2]
```diff
@@ -0,0 +1,41 @@
+FIRST I GOT TO SETUP THE MATERIALS CONFIGURATION UPDATE (that will be hard)
```
Comment (anttiviljakainen) [inline muikku-core-plugins/src/main/resources/META-INF/resources/jsf/workspace/structure.txt:1]: Is this file relevant to the functionality of the Thing?

File: muikku-core-plugins/src/main/resources/fi/otavanopisto/muikku/plugins/guider/GuiderJsPluginMessages_fi.properties [hunk 3]
```diff
@@ -57,7 +59,9 @@ plugin.guider.user.details.workspaces = Kurssit
 plugin.guider.user.details.statistics = Tilastot
 plugin.guider.user.details.files = Tiedostot
 plugin.guider.user.details.files.hint = Lis�� liitetiedosto klikkaamalla t�st� tai raahamalla tiedosto t�h�n laatikkoon
+plugin.guider.user.details.files.fileFieldUpload.fileSizeTooLarge = Tiedoston koko on liian suuri. Suurin sallittu tiedostokoko on 10MB.
 plugin.guider.user.details.files.empty = Ei tiedostoja
+plugin.guider.user.details.files.uploading = Ladataan
```
Comment (anttiviljakainen) [inline muikku-core-plugins/src/main/resources/fi/otavanopisto/muikku/plugins/guider/GuiderJsPluginMessages_fi.properties:64]: files.uploading --- file.remove order reversed compared to english locale

File: muikku-core-plugins/src/main/resources/fi/otavanopisto/muikku/plugins/workspace/WorkspacePluginMessages_fi.properties [hunk 4]
```diff
@@ -88,4 +88,4 @@ plugin.workspace.management.workspaceProducersSectionTitle = Ty
 plugin.workspace.management.workspaceButtons.save = Tallenna muutokset
 plugin.workspace.management.chatSettings=Ty�tilan chatin asetukset
 plugin.workspace.announcer.pageTitle = Ty�tilan tiedotteet
-plugin.workspace.management.cannotRemoveAnswers = Toiminto estetty, koska se poistaisi teht�v�vastauksia. Ota yhteytt� yll�pit�j��n.
\ No newline at end of file
+plugin.workspace.management.cannotRemoveAnswers = Toiminto estetty, koska se poistaisi teht�v�vastauksia. Ota yhteytt� yll�pit�j��n.
```
Comment (anttiviljakainen) [inline muikku-core-plugins/src/main/resources/fi/otavanopisto/muikku/plugins/workspace/WorkspacePluginMessages_fi.properties:91]: Locale file doesn't include all the rows in english version.

File: muikku-core-plugins/src/main/resources/META-INF/resources/jsf/workspace/structure.txt (hunk 2 above)
Comment (jjlankinen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/jsf/workspace/structure.txt:1]: I would think not. Deleted it.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/base/mixins.scss [hunk 5]
```diff
@@ -8,25 +8,122 @@
   #{$property}: $values;
 }
 
+@mixin no-scrollbars {
+  overflow: -moz-scrollbars-none;
+  -ms-overflow-style: none;
+  scrollbar-width: none;
```
[the remark this answers is not in the document]
Comment (jjlankinen) [reply muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/sass/base/mixins.scss:14]: scrollbar-width is level 1 draft, lets pretend it's official for now.

File: muikku-rest/src/main/java/fi/otavanopisto/muikku/rest/PermissionRESTService.java [hunk 6]
```diff
@@ -30,24 +46,157 @@
...
+    WorkspaceEntity workspaceEntity = workspaceController.findWorkspaceEntityById(workspaceEntityId);
+    if (!sessionController.hasPermission(MuikkuPermissions.WORKSPACE_MANAGEWORKSPACESETTINGS, workspaceEntity)) {
+      return Response.status(Status.FORBIDDEN).build();
+    }
+    
+    List<UserGroupEntity> userGroupEntities;
+    
+    String permissionGroupIds = pluginSettingsController.getPluginSetting("workspace", "permission-group-ids");
+    if (permissionGroupIds == null) {
+      userGroupEntities = userGroupEntityController.listUserGroupEntities();
```
Comment (anttiviljakainen) [inline muikku-rest/src/main/java/fi/otavanopisto/muikku/rest/PermissionRESTService.java:103]: Needs some rethinking as this f.ex. has no organization restriction.

File: muikku-rest/src/main/java/fi/otavanopisto/muikku/rest/PermissionRESTService.java [hunk 7]
```diff
@@ -30,24 +46,157 @@
...
+  @PUT
+  @Path ("/workspaceSettings/{WORKSPACEENTITYID}/userGroups/{USERGROUPID}")
+  @RESTPermit(handling = Handling.INLINE, requireLoggedIn = true)
+  public Response setWorkspaceSettingsUserGroup(@PathParam("WORKSPACEENTITYID") Long workspaceEntityId,
+      @PathParam("USERGROUPID") Long userGroupEntityId, WorkspaceSettingsUserGroup payload) {
+    if (!Objects.equals(workspaceEntityId, payload.getWorkspaceEntityId()) || !Objects.equals(userGroupEntityId, payload.getUserGroupEntityId())) {
+      return Response.status(Status.BAD_REQUEST).build();
+    }
+    
+    UserGroupEntity userGroupEntity = userGroupEntityController.findUserGroupEntityById(userGroupEntityId);
```
Comment (anttiviljakainen) [inline muikku-rest/src/main/java/fi/otavanopisto/muikku/rest/PermissionRESTService.java:160]: This needs to be restricted to the groups in organizations the user has access to.

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/material/rest/HtmlMaterialRESTService.java [hunk 8]
```diff
@@ -125,10 +125,43 @@ public Response findMaterial(@PathParam("id") Long id, @QueryParam ("revision")
...
     }
   }
-  
+
+  @PUT
+  @Path("/{id}/content")
+  @RESTPermit (handling = Handling.INLINE, requireLoggedIn = true)
+  public Response updateMaterialContent(@PathParam("id") Long id, HtmlRestMaterialContent entity) {
+    
+    if (!sessionController.hasEnvironmentPermission(MuikkuPermissions.MANAGE_MATERIALS)) {
```
Comment (anttiviljakainen) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/material/rest/HtmlMaterialRESTService.java:134]: Is MANAGE_MATERIALS restrictive enough?
